bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Epoch.scala

```diff
@@ -1,18 +1,116 @@
 package edu.gemini.spModel.core
 
-import scalaz._, Scalaz._
+import java.time.ZoneOffset.UTC
+import java.time._
 
-/** Epoch in Gregorian (?) years. */
-final case class Epoch(year: Double)
+import scalaz._
+import Scalaz._
 
+/**
+ * An epoch, the astronomer's equivalent of `Instant`, based on a fractional year in some temporal
+ * scheme (Julian, in this case) that determines year zero and the length of a year. The only
+ * meaningful operation for an `Epoch` is to ask the elapsed epoch-years between it and some other
+ * point in time. We need this for proper motion corrections because velocities are measured in
+ * motion per epoch-year.
+ *
+ * @see The Wikipedia [[https://en.wikipedia.org/wiki/Epoch_(astronomy) article]]
+ */
+final case class Epoch(year: Double) {
+  /** Offset in epoch-years from this `Epoch` to the given `Instant`. */
+  def untilInstant(i: Instant): Double =
+    untilLocalDateTime(LocalDateTime.ofInstant(i, ZoneOffset.UTC))
+
+  /** Offset in epoch-years from this `Epoch` to the given `LocalDateTime`. */
+  def untilLocalDateTime(ldt: LocalDateTime): Double =
+    untilJulianDay(Epoch.toJulianDay(ldt))
+
+  /** Offset in epoch-years from this `Epoch` to the given fractional Julian day. */
+  def untilJulianDay(jd: Double): Double =
+    Epoch.JulianYearBasis + (jd - Epoch.JulianBasis) / Epoch.JulianLengthOfYear
```

Comment:
See the [sanity checks](https://github.com/gemini-hlsw/gsp-math/blob/master/modules/tests/shared/src/test/scala/gsp/math/EpochSpec.scala) in `gsp-math`.  We should do the same I think.